examples/eks_managed_node_group/main.tf

```diff
@@ -388,6 +389,9 @@ module "disabled_eks_managed_node_group" {
   source = "../../modules/eks-managed-node-group"
 
   create = false
+
+  # Hard requirement
+  cluster_service_cidr = ""
```

Comment:
This is the first time we require something to be set for "disabled" modules, as far as I see. Can we still make it optional to be consistent?

Reply:
I've come up with a way to make this only required when `create = true` using `precondition` on a `null_resource` that does nothing (but enforce the check) https://github.com/terraform-aws-modules/terraform-aws-eks/pull/2963/files#diff-87747f89f95b121dfc7a5f6e40ee0a56f1f98449581c02bd34f2171210c2f14dR1-R12

When `create = true` and you do not provide a `cluster_service_cidr` you now get this error which is what we (I) want since it is required when creating a nodegroup:
![image](https://github.com/terraform-aws-modules/terraform-aws-eks/assets/10913471/32910364-ebeb-4545-b4ce-d479a18b091d)

I did have to bump the Terraform MSV from `1.3.0` to `1.3.2` to pick up this patch https://github.com/hashicorp/terraform/issues/31846